src/Ionide.ProjInfo/Library.fs

```diff
@@ -167,7 +168,7 @@ module Init =
 
         match System.Environment.GetEnvironmentVariable "DOTNET_HOST_PATH" with
         | null
-        | "" -> Environment.SetEnvironmentVariable("DOTNET_HOST_PATH", Paths.dotnetRoot.FullName)
+        | "" -> Environment.SetEnvironmentVariable("DOTNET_HOST_PATH", dotnetExe.FullName)
```

Comment:
Hmmm, My problem in #123 was more that this environment variable was being set within the process to an incorrect value, and then never unset.  This caused FSharp.Compiler.Service to do incorrect things.

So is the fix here that `dotnetExe` will now be a good value? 

But even if it's a good value, I feel there should be an unset of this environment variable after sub-process launch, in the cases where it gets set.  I had to add an explicit unset here for example: https://github.com/fsprojects/FSharp.Formatting/blob/main/src/fsdocs-tool/BuildCommand.fs#L851

Reply:
It turns out that we do need a value of `DOTNET_HOST_PATH` set, because the F# targets rely on it to set the fsc binary to invoke to `fsc.dll` instead of `fsc.exe`. This logic is [here](https://github.com/dotnet/fsharp/blob/main/src/fsharp/FSharp.Build/Microsoft.FSharp.NetSdk.props#L59-L67). Without this value set, our tests start failing left and right.  I could probably try a version where we set these msbuild properties in this library explicitly, but I worry about the fragility of that, as these properties are subject to change at any time, while DOTNET_HOST_PATH is documented.

Another option would be for the `init` function to return an `IDisposable` that you could dispose to  'pop' the changes to the environment variables, but that's a bit of a foot-gun - if disposed then everything in this library becomes unusable until `init` is called again, so you'd want some sort of scoping on the client side.